### PR TITLE
JAMES-2787 Add depolyement test for IMAP SELECT

### PR DESCRIPTION
**Pull request by blackheaven**

### CODE REVIEW

File: mpt/impl/imap-mailbox/external-james/src/test/resources/org/apache/james/imap/scripts/ValidateDeployment.test [hunk 1]
```diff
@@ -236,6 +236,7 @@ S: A10 OK FETCH completed\.
 
 C: A11 LIST "" "*"
 SUB {
+S: \* LIST \(\\HasNoChildren\) "\." "fetchTest"
```
Comment (chibenwa) [inline mpt/impl/imap-mailbox/external-james/src/test/resources/org/apache/james/imap/scripts/ValidateDeployment.test]: Does this means you enforce the order the test methods should be runned in?

**Comment by trantienduchn** [comment]
read it

**Comment by Arsnael** [comment]
Read it too

File: mpt/impl/imap-mailbox/external-james/src/test/resources/org/apache/james/imap/scripts/ValidateDeployment.test (hunk 1 above)
Comment (blackheaven) [reply mpt/impl/imap-mailbox/external-james/src/test/resources/org/apache/james/imap/scripts/ValidateDeployment.test]: sadly, do you have any idea about that?

Comment (chibenwa) [reply mpt/impl/imap-mailbox/external-james/src/test/resources/org/apache/james/imap/scripts/ValidateDeployment.test]: Can we delete the fetchTest mailbox at the end of the tests, at the very least?

**Comment by blackheaven** [comment]
```
[ERROR]   RabbitMQEventBusTest$LifeCycleTest$SingleEventBus$DispatchingWhenNetWorkIssue.dispatchShouldWorkAfterNetworkIssuesForOldRegistration:333 » ConditionTimeout
```

test this please

File: mpt/impl/imap-mailbox/external-james/src/test/resources/org/apache/james/imap/scripts/CheckFetch.test [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+C: Test content
+S: \* 1 EXISTS
+S: \* 1 RECENT
+S: A003 OK (\[.+\] )?APPEND completed\.
+
+C: A10 FETCH *:* (BODY UID)
+S: \* 1 FETCH \(BODY \("TEXT" "PLAIN" \("charset" "us-ascii"\) NIL NIL "7BIT" 12 0\) UID 1\)
+S: A10 OK FETCH completed\.
+
+C: 20 DELETE fetchTest
```
Comment (mbaechler) [inline mpt/impl/imap-mailbox/external-james/src/test/resources/org/apache/james/imap/scripts/CheckFetch.test]: could you use consistent tags?

Comment (blackheaven) [reply mpt/impl/imap-mailbox/external-james/src/test/resources/org/apache/james/imap/scripts/CheckFetch.test]: 10?

Comment (chibenwa) [reply mpt/impl/imap-mailbox/external-james/src/test/resources/org/apache/james/imap/scripts/CheckFetch.test]: A11

Comment (blackheaven) [reply mpt/impl/imap-mailbox/external-james/src/test/resources/org/apache/james/imap/scripts/CheckFetch.test]: thanks

Comment (mbaechler) [reply mpt/impl/imap-mailbox/external-james/src/test/resources/org/apache/james/imap/scripts/CheckFetch.test]: Actually, I didn't repeat myself, but all tags should be consistent, which is not the case.

**Comment by blackheaven** [comment]
test this please

**Comment by blackheaven** [comment]
```
Caused by: java.lang.IllegalStateException: no known node for /home/jenkins/build/workspace/726
```

can be considered :green_apple: ?

**Comment by blackheaven** [comment]
```

[fdd085f8d527671ffad4e1ee0f9bcbd534cb3811] LastClientMsg: c SEARCH FLAGGED
[fdd085f8d527671ffad4e1ee0f9bcbd534cb3811] Expected: '\* SEARCH 1 2 3 4'
[fdd085f8d527671ffad4e1ee0f9bcbd534cb3811] Actual   : '* SEARCH'
[fdd085f8d527671ffad4e1ee0f9bcbd534cb3811] 	at org.apache.james.mpt.imapmailbox.external.james.DeploymentValidation.validateDeployment(DeploymentValidation.java:81)
[fdd085f8d527671ffad4e1ee0f9bcbd534cb3811] 
```

**Comment by blackheaven** [comment]
test this please

**Comment by blackheaven** [comment]
test this please

**Comment by blackheaven** [comment]
```

[55af4b985ad38ca2e875e4617cb31179a493b454] Location: /org/apache/james/imap/scripts/ValidateDeployment.test:499
[55af4b985ad38ca2e875e4617cb31179a493b454] LastClientMsg: c SEARCH FLAGGED
[55af4b985ad38ca2e875e4617cb31179a493b454] Expected: '\* SEARCH 1 2 3 4'
[55af4b985ad38ca2e875e4617cb31179a493b454] Actual   : '* SEARCH'
[55af4b985ad38ca2e875e4617cb31179a493b454] 	at org.apache.james.mpt.imapmailbox.external.james.DeploymentValidation.validateDeployment(DeploymentValidation.java:81)
[55af4b985ad38ca2e875e4617cb31179a493b454] 
[55af4b985ad38ca2e875e4617cb31179a493b454] [INFO] 
[55af4b985ad38ca2e875e4617cb31179a493b454] [INFO] Results:
[55af4b985ad38ca2e875e4617cb31179a493b454] [INFO] 
[55af4b985ad38ca2e875e4617cb31179a493b454] [ERROR] Errors: 
[55af4b985ad38ca2e875e4617cb31179a493b454] [ERROR]   DeploymentValidation.validateDeployment:81 » InvalidServerResponse 
[55af4b985ad38ca2e875e4617cb31179a493b454] Location: ...
```

test this please

File: mpt/impl/imap-mailbox/external-james/src/test/java/org/apache/james/mpt/imapmailbox/external/james/DeploymentValidation.java [hunk 3]
```diff
@@ -81,6 +81,11 @@ public void validateDeployment() throws Exception {
         simpleScriptedTestProtocol.run("ValidateDeployment");
     }
 
+    @Test
+    public void selectWithExistingMessages() throws Exception {
+        simpleScriptedTestProtocol.run("SelectWithExistingMessages");
```
Comment (mbaechler) [inline mpt/impl/imap-mailbox/external-james/src/test/java/org/apache/james/mpt/imapmailbox/external/james/DeploymentValidation.java]: as far as I understand, there's no existing message when calling SELECT.

**Pull request merged**